### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by rchristie**

### CODE REVIEW

**Review by mahyar-osn — APPROVED** [review]
Looks great. I tested the code and works well. I had a few very minor comments.

File: src/dataembedder/dataembedder.py [hunk 1]
```diff
@@ -0,0 +1,870 @@
...
+        """
+        Get number of objects in data group of name of its highest dimension.
+        :param groupName: Name of the group
+        :return: Size > 0, or -1 if group not found.
+        """
+        groupDict = self._groupData.get(groupName)
+        if groupDict:
+            return groupDict[self._sizeToken]
+        print("DataEmbedder getGroupSize: no group of name " + str(groupName), file=sys.stderr)
+        return -1
```
[the remark this answers is not in the document]
Comment (rchristie) [reply src/dataembedder/dataembedder.py:724]: Return type should have been int. Fixed.

File: tests/test_dataembedder.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import unittest
+from opencmiss.utils.zinc.field import get_group_list
+from opencmiss.utils.zinc.finiteelement import evaluate_field_nodeset_range
+from opencmiss.zinc.field import Field
+from dataembedder.dataembedder import DataEmbedder
+
+here = os.path.abspath(os.path.dirname(__file__))
+
+
+def assertAlmostEqualList(testcase, actualList, expectedList, delta):
```
[the remark this answers is not in the document]
Comment (rchristie) [reply tests/test_dataembedder.py:11]: Fixed.

File: src/dataembedder/dataembedder.py [hunk 3]
```diff
@@ -0,0 +1,870 @@
...
+        """
+        Get highest dimension in objects of data group of name.
+        :param groupName: Name of the group
+        :return: Dimension >= 0, or -1 if group not found.
+        """
+        groupDict = self._groupData.get(groupName)
+        if groupDict:
+            return groupDict[self._dimensionToken]
+        print("DataEmbedder getGroupDimension: no group of name " + str(groupName), file=sys.stderr)
+        return 0
```
[the remark this answers is not in the document]
Comment (rchristie) [reply src/dataembedder/dataembedder.py:682]: Changed return type to int.

**Pull request merged**